### PR TITLE
Main Screen View Model Tests Written

### PR DESCRIPTION
**Pull request by Manik2708**
**What kind of change does this PR introduce?**
1) Improved the code coverage of the `MainScreenViewModel.dart`
2) Added multiple tests to improve the effectiveness of tests
3) Some of tests were not matching/expecting anything, improved those tests.

**Issue Number:**

Fixes #1121

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Manik2708** [comment]
Sorry for the inconvenience caused! I have to close the previous PR due to large number of conflicts. Please @noman2002 @palisadoes @literalEval @Ayush0Chaudhary review.

**Comment by palisadoes** [comment]
@Manik2708 

There has been an increasing trend with PRs where they are closed without explanation even when they are made against the develop branch as expected. 

This is a real tax on all our time. 

1) The discussion history gets reset with the new PR
2) We have to find new assigned reviewers
3) Tracking becomes harder
4) The reviewers work is easier to get lost.

We have made many warnings about this with little progress.

With immediate effect we will unassign assignees whenever this happens against the develop branch.

I don't want to speculate on the reason, but we cannot let this continue.

**Comment by literalEval** [comment]
@Manik2708 

1. Why are the files regenerated ?
2. Since you have modified the files (that too for tests, specifically), you are supposed to fix the side stuff too. Most of the documentation is wrong in the file, there are multiple print statements, and unused code is commented instead of being removed.

Writing tests does not simply mean covering lines in codecov. Please help us in improving code quality as well. Moreover, a line is still uncovered (look at codecov comment).

**Comment by Manik2708** [comment]
> @Manik2708
> 
> 1. Why are the files regenerated ?
> 2. Since you have modified the files (that too for tests, specifically), you are supposed to fix the side stuff too. Most of the documentation is wrong in the file, there are multiple print statements, and unused code is commented instead of being removed.
> 
> Writing tests does not simply mean covering lines in codecov. Please help us in improving code quality as well. Moreover, a line is still uncovered (look at codecov comment).

I have not edited the main file, due to large number of conflicts I just copied the file in new file, with an additional parameter in the function`tourHomeTargets`. I am still unknown why GitHub have counted the rest lines in this commit. I tried my best to reduce the number of changes in this PR but failed. If you are saying that there are problems in main file also then I can fix them. Regarding the effectiveness of Tests, yes previously written tests, which were having 97% coverage were just covering the lines, in this PR I actually made them effective. I also highlighted this problem in issue #1121.

**Comment by Azad99-9** [comment]
@Manik2708 please resolve conflicts.

**Comment by Manik2708** [comment]
> @Manik2708 please resolve conflicts.

Can you help me out, as to resolve conflicts I have to pull the latest branch but for that I have to discard my commits that means close this PR. How can I deal with this?

**Comment by Dante291** [comment]
> > @Manik2708 please resolve conflicts.
> 
> Can you help me out, as to resolve conflicts I have to pull the latest branch but for that I have to discard my commits that means close this PR. How can I deal with this?

Push the changes you've made (if any) then take a latest pull, this way the changes you've made and latest upstream you've merged will show all the  conflicts, resolve these conflicts manually and then push the files again

**Comment by Manik2708** [comment]
> > > @Manik2708 please resolve conflicts.
> > 
> > 
> > Can you help me out, as to resolve conflicts I have to pull the latest branch but for that I have to discard my commits that means close this PR. How can I deal with this?
> 
> Push the changes you've made (if any) then take a latest pull, this way the changes you've made and latest upstream you've merged will show all the conflicts, resolve these conflicts manually and then push the files again

Oh! Thanks a lot, I was missing that "Manually fixing conflicts!" Thanks again

**Comment by Manik2708** [comment]
@noman2002 , @literalEval , @Ayush0Chaudhary So sorry for these delays and all the mess I created. I learnt a lot and was busy with my college. Please review!

**Comment by Dante291** [comment]
> > > > @Manik2708 please resolve conflicts.
> > > 
> > > 
> > > Can you help me out, as to resolve conflicts I have to pull the latest branch but for that I have to discard my commits that means close this PR. How can I deal with this?
> > 
> > 
> > Push the changes you've made (if any) then take a latest pull, this way the changes you've made and latest upstream you've merged will show all the conflicts, resolve these conflicts manually and then push the files again
> 
> Oh! Thanks a lot, I was missing that "Manually fixing conflicts!" Thanks again

All cool bro, also you could resolve conflicts in github editor too

File: test/view_model_tests/main_screen_view_model_test.dart [hunk 1]
```diff
@@ -1,686 +1,927 @@
...
+    //             onPressed: () {
+    //               model2.tourProfile();
+    //               model2.showTutorial(onClickTarget: (targetFocus) {
+    //                 return TargetFocus();
+    //               }, onFinish: () {
+    //                 return TargetFocus();
+    //               });
+    //             },
+    //           ),
+    //         );
```
Comment (Ayush0Chaudhary) [inline test/view_model_tests/main_screen_view_model_test.dart:786]: remove commented out code

**Review by Ayush0Chaudhary — COMMENTED** [review]
rmv commented out code, everything else looks good

File: test/view_model_tests/main_screen_view_model_test.dart (hunk 1 above)
Comment (Manik2708) [reply test/view_model_tests/main_screen_view_model_test.dart]: Done

**Review by noman2002 — APPROVED** [review]
Looks fine.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**